### ISSUE TITLE
Feature Request: Generate app apk and test apk if not found

### ISSUE THREAD

**Issue by tir38** [runningcode/fladle#63]
Right now I have to build app and test apks before running flank:

`./gradlew clean app:assembleDebug app:assembleDebugAndroidTest app:runFlank --no-parallel`

I know that you already added "auto detect apks" in #1. It would be great if Fladle was one step smarter and I could just call `./gradlew app:runFlank` and have it work out all the magic.

**Comment by runningcode** [comment]
Thanks for the feature request. I need to think about this and how to find the correct task for all projects.
As a workaround, you can make the runFlank task depend on the `assembleDebug` and `assembleDebugAndroidTest`.

**Comment by runningcode** [comment]
Something like this should work
```
tasks.named("execFlank").configure {
  dependsOn("assembleDebugAndroidTest")
}
```

**Comment by pjakubczyk** [comment]
I started to have same problem when migrated to AGP3.5 and Gradle 5.6
Your snipped solved it but maybe it'd be great to mention that in Readme ?
I guess it's going to be more common

**Comment by tir38** [comment]
I assume I have to do this after project evaluation

```
afterEvaluate {
    tasks.named("execFlank").configure {
        dependsOn("assembleDebugAndroidTest")
    }
}
```

**Comment by pjakubczyk** [comment]
@tir38 I did it exactly like you wrote

**Comment by runningcode** [comment]
There's two main things to consider:
1. This would be a breaking change, it would add new task dependency which could break some CI setups. Those with multiple stages where the debug apk and test apk are generated in one stage and copied to another. This would mean that the app is recompiled even though the output apk is already there.
I could hide this behind an extension config which is disabled by default to solve this.
2. Is determining which variant and test variant to depend on. I could hardcode to "assembleDebug" and "assembleAndroidDebugTest" to start. Where the dependency is only created if those tasks exist.

How does that sound? Any other thoughts or ideas?

**Comment by pjakubczyk** [comment]
In case of somebody has problems with settings the dependency I found that using the `execFlank` is sometimes causing problems. I guess because `execFlank ` and `assembleDebugAndroidTest` are completely independent so Gradle will run them in parallel.

Anyways I solved the problem by setting dependency on `writeConfigProps`
```
afterEvaluate {
    tasks.named("writeConfigProps").configure {
        dependsOn("assembleDebugAndroidTest")
    }
}
```

**Comment by runningcode** [comment]
Thanks for the update. Are you saying that the solution in this [comment](https://github.com/runningcode/fladle/issues/63#issuecomment-524586097) and [readme](https://github.com/runningcode/fladle#error-apk-file-not-found) do not work?
Were you trying with `execFlank` or `runFlank`?

**Comment by pjakubczyk** [comment]
or maybe setting new dependencies on a task ain't that accurate ?

**Comment by runningcode** [comment]
What was the exact code you tried?

**Comment by pjakubczyk** [comment]
actually I have a set of configurations
```
afterEvaluate {
    ["runFlankMockAPI", "runFlankSmallMedium", "runFlankEndToEnd", "runFlankAdHoc"].each { fladleTask ->
        tasks.named(fladleTask).configure {
            dependsOn([variant.assemble, "assemble${sourceName}AndroidTest"])
        }
    }
}
```
and while doing this it was like a race condition to have the both apks ready :/
I changed it to
```
afterEvaluate {
    ["writeConfigPropsMockAPI", "writeConfigPropsSmallMedium", "writeConfigPropsEndToEnd", "writeConfigPropsAdHoc"].each { fladleTask ->
        tasks.named(fladleTask).configure {
            dependsOn(variant.assemble, variant.testVariant.assemble)
        }
    }
}
```

**Comment by runningcode** [comment]
Hey, thanks for the sample code. I apologize for the confusion. The `runFlank` tasks and other do not actually perform any work. They all depend on the `execFlank` task which actually performs the work. Thus adding a dependency to the `runFlank` task will not do anything.

Can you try the following code?
```
afterEvaluate {
    ["execFlankMockAPI", "execFlankSmallMedium", "execFlankEndToEnd", "execFlankAdHoc"].each { fladleTask ->
        tasks.named(fladleTask).configure {
            dependsOn([variant.assemble, "assemble${sourceName}AndroidTest"])
        }
    }
}
```

**Comment by pjakubczyk** [comment]
Yep that works. 
Set dependencies on `execFlank*` not on `runFlank*`.
Thanks!

**Comment by tir38** [comment]
I don't know why my above post was working for us (shrug). But for whatever reason I now need to build app apk and test apk (which makes total sense)

```
afterEvaluate {
    tasks.named("execFlank").configure {
        dependsOn("assemble[BUILDTYPE]") // had to add this; not sure why it *ever* worked w/out??
        dependsOn("assemble[BUILDTYPE][AndroidTest")
    }
}
```
